src/python/pants/cache/restful_artifact_cache.py

```diff
@@ -108,13 +108,23 @@ def _request(self, method, cache_key, body=None):
       logger.debug('Sending {0} request to {1}'.format(method, url))
       try:
         if 'PUT' == method:
-          response = session.put(url, data=body, timeout=self._write_timeout_secs)
+          response = session.put(url,
```

Comment:
NB: PUT, GET and DELETE already followed redirects by default - I'm just making this explicit here.